src/gui/folder.h

```diff
@@ -353,6 +353,7 @@ private slots:
     bool _csyncUnavail;
     bool _proxyDirty;
     QPointer<RequestEtagJob> _requestEtagJob;
+    QScopedPointer<QTcpSocket> _webSocket;
```

Comment:
QSslSocket?

Reply:
Guide me - can QSslSocket also establish a non-ssl connection?

Reply:
https://code.woboq.org/qt5/qtbase/src/network/ssl/qsslsocket.cpp.html#_ZN10QSslSocket22connectToHostEncryptedERK7QStringt6QFlagsIN9QIODevice12OpenModeFlagEEN15QAbstractSocket20NetworkLayerProtocolE

Reply:
https://code.woboq.org/qt5/qtbase/src/network/ssl/qsslsocket.cpp.html#_ZN10QSslSocket13connectToHostERK7QStringt6QFlagsIN9QIODevice12OpenModeFlagEEN15QAbstractSocket20NetworkLayerProtocolE

Reply:
You can use one of those depending on the account URL http vs https or how would this work? I wonder if the websocket URL should come from capabilities though so the server admin can scale it differently..

Things to keep in mind is the sslErrors signal. We have a ```AbstractSslErrorHandler``` that you can use.

The question is if websockets could return different ssl errors or not. Maybe as first implementation you assume they return the same error as the normal account WebDAV calls ..

Reply:
The server would expose the websocket url via a capability. The Url will either be ws:// or wss://. 

So depending on using ssl or not I either call connectToHost or connectToHostEncrypted ?

Reply:
I'll postpone the ssl topic - websockets with self signed certificates are not working.
Hard to test at the moment.